src/metabase/db/data_migrations.clj

```diff
@@ -369,6 +369,19 @@
                                  [:like
                                   :dashcard.visualization_settings "%\"click_link_template\":%"]]})))
 
+(defmigration ^{:author "jeff303" :added "0.43.0"} remove-bigquery-driver
+  (doseq [{:keys [:details] :as bigquery-db} (db/select Database :engine "bigquery")]
+    (when-not (empty? (filter some? ((juxt :auth-code :client-id :client-secret) details)))
+      (log/errorf (str "Database ID %d used the :bigquery driver with OAuth style authentication scheme, which cannot"
+                       " be automatically migrated to :bigquery-cloud-sdk; leaving authentication options nil in the"
+                       " db details, which must be corrected by an admin (by setting a service-account-json), so that"
+                       " sync and query can work again")
+                  (u/the-id bigquery-db)))
+    (let [migrated-details (-> (assoc details :engine :bigquery-cloud-sdk) ; switch to new driver
```

Comment:
`engine` is a separate top-level column in `Database`. It's not inside `details`

Reply:
Oops, you'd think I would remember this by now.